src/python/zquantum/qcbm/ansatz.py

```diff
@@ -254,4 +259,63 @@ def get_number_of_parameters_by_layer(self) -> np.ndarray:
                 # A normal single qubit rotation layer
                 num_params_by_layer.append(self.number_of_qubits * 2)
 
-        return np.asarray(num_params_by_layer)
\ No newline at end of file
+        return np.asarray(num_params_by_layer)
+
+    def to_dict(self):
+        """Creates a dictionary representing a QCBM ansatz.
+
+        Returns:
+            dictionary (dict): the dictionary
+        """
+        dictionary = {
+            "schema": ANSATZ_SCHEMA,
+            "number_of_layers": self.number_of_layers,
+            "number_of_qubits": self.number_of_qubits,
+            "topology": self.topology,
+        }
+
+        return dictionary
+
+
+def save_qcbm_ansatz_set(qcbm_ansatz_set: List[QCBMAnsatz], filename: str) -> None:
+    """Save a set of qcbm_ansatz to a file.
+
+    Args:
+        qcbm_ansatz_set (list): a list ansatz to be saved
+        file (str): the name of the file
+    """
+    dictionary = {
+        "schema": ANSATZSET_SCHEMA,
+        "qcbm_ansatz": [ansatz.to_dict() for ansatz in qcbm_ansatz_set]
+    }
+
+    with open(filename, "w") as f:
+        f.write(json.dumps(dictionary, indent=2))
+
+
+def load_qcbm_ansatz_set(file: str) -> List[QCBMAnsatz]:
+    """Load a list of qcbm_ansatz from a json file using a schema.
+
+    Arguments:
+        file (str): the name of the file
+
+    Returns:
+        object: a list of qcbm_ansatz loaded from the file
+    """
+    if isinstance(file, str):
+        with open(file, "r") as f:
+            data = json.load(f)
+    else:
+        data = json.load(file)
+
+    qcbm_ansatz_set = []
+    for item in data["qcbm_ansatz"]:
+        qcbm_ansatz_set.append(
+            QCBMAnsatz(
+                number_of_layers=item["number_of_layers"],
+                number_of_qubits=item["number_of_qubits"],
+                topology=item["topology"],
+            )
```

Comment:
This certainly works, but it might make more sense to create a `from_dict` method for the `QCBMAnsatz` class. Then you could write this as `QCBMAnsatz.from_dict(item)`. But this is not necessary.

Reply:
Done!